src/Futhark/MonadFreshNames.hs

```diff
@@ -76,6 +77,16 @@ modifyNameSource m = do
   putNameSource src'
   pure x
 
+-- | Run a computation needing a fresh name source and returning a new
+-- one, using 'getNameSource' and 'putNameSource' before and after the
+-- computation.
+modifyNameSourceIO :: (MonadFreshNames m, MonadIO m) => (VNameSource -> IO (a, VNameSource)) -> m a
```

Comment:
???

Reply:
This was a remnant of the z3 dependency. I've removed it.